driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -143,29 +143,25 @@ static void validateCollationAndWriteConcern(final ConnectionDescription connect
         }
     }
 
-    static void validateArrayFiltersAndWriteConcern(final ConnectionDescription connectionDescription,
-                                                    final List<BsonDocument> arrayFilters, final WriteConcern writeConcern) {
-        if (arrayFilters != null) {
-            if (serverIsLessThanVersionThreeDotSix(connectionDescription)) {
-                throw new IllegalArgumentException(format("Array filters not supported by wire version: %s",
-                        connectionDescription.getMaxWireVersion()));
-            } else if (!writeConcern.isAcknowledged()) {
-                throw new MongoClientException("Specifying array filters with an unacknowledged WriteConcern is not supported");
-            }
+    private static void validateArrayFilters(final ConnectionDescription connectionDescription,
+                                             final List<BsonDocument> arrayFilters, final WriteConcern writeConcern) {
+        if (serverIsLessThanVersionThreeDotSix(connectionDescription)) {
+            throw new IllegalArgumentException(format("Array filters not supported by wire version: %s",
+                    connectionDescription.getMaxWireVersion()));
+        } else if (!writeConcern.isAcknowledged()) {
+            throw new MongoClientException("Specifying array filters with an unacknowledged WriteConcern is not supported");
         }
     }
 
-    static void validateHintAndWriteConcern(final ConnectionDescription connectionDescription, final Bson hint, final String hintString,
-                                            final WriteConcern writeConcern) {
-        if (hint != null || hintString != null) {
-            if (serverIsLessThanVersionThreeDotFour(connectionDescription)) {
-                throw new IllegalArgumentException(format("Hint not supported by wire version: %s",
-                        connectionDescription.getMaxWireVersion()));
-            } else if (!writeConcern.isAcknowledged() && serverIsLessThanVersionFourDotTwo(connectionDescription)) {
-                throw new MongoClientException(
-                        format("Specifying hints with an unacknowledged WriteConcern is not supported by wire version: %s",
-                                connectionDescription.getMaxWireVersion()));
-            }
+    private static void validateHint(final ConnectionDescription connectionDescription, final Bson hint, final String hintString,
```

Comment:
The hint and hintString arguments are no longer used.

Reply:
Removed.